src/scene/Entity.hpp

```diff
@@ -14,37 +14,68 @@
 
 #pragma once
 
+#include <utility>
+
+#include <APIObject.hpp>
+#include <RGLFields.hpp>
 #include <scene/Scene.hpp>
 #include <scene/Mesh.hpp>
-#include <APIObject.hpp>
-#include <utility>
 #include <math/Mat3x4f.hpp>
 
-
+/**
+ * Entity represents an instance of a mesh on the scene, with attributes such as transform, id, intensity texture, etc.
```

Comment:
This description is misleading. 
More of a "tuple". 
Phrasing this way suggests that mesh has the following attributes which is not true.

Reply:
Good catch, fixed